src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmersSpark.java

```diff
@@ -31,7 +32,9 @@
     private static final long serialVersionUID = 1L;
     @VisibleForTesting static final Long MAX_KMER_FREQ = 3L;
     private static final int REF_RECORD_LEN = 10000;
-    private static final int CHUNK_SIZE = 3000000;
+    // to give a task size of 100Kb -- the max size recommended in the Spark documentation
+    private static final int REF_RECORDS_PER_PARTITION = 102400 / REF_RECORD_LEN;
```

Comment:
It seems to me that part of what's necessary is not purely technical.  Our future clients will be deploying our software on their own cloud-based or local Spark clusters.  Either we design our software to detect and react to the available memory in the environment, or we must specify a minimum practical environment.  I think our Spark cluster is very unusual in having a small number of beefy machines.  More likely, I'd guess, would be a large number of machines with limited memory.  Will our software (not just this silly little application) behave?